### PR TITLE
Merge hometown's norss setting into glitch

### LINKED ISSUE #1936: [feature request] allow user to toggle RSS feed of their posts
### Pitch

currently every user has an automatic RSS feed containing their unlisted and public posts, i think it would be good if there was some account setting that allowed the user to turn this off.

### Motivation

while unlisted/public posts are not hidden from other users, i can see where providing that data in an easy to scrape format (RSS feeds) would help provide new and faster methods for harassing users.

### PR DESCRIPTION
**Pull request by prplecake**
Adding a `norss` user preference

(cherry picked from commit 29eb83d073410919b17a3ef8a6eaf6c492bb4284)

fixes #1936 

This adds a user preference, and when that preference is `true`, the RSS feed still exists but is empty, containing no statuses. This also removes the 'link alternative' meta tag.

This also adds an admin setting that can enable the option for all users.

### CODE REVIEW

File: app/views/settings/preferences/other/show.html.haml [hunk 1]
```diff
@@ -10,13 +10,19 @@
...
   .fields-group
     = f.input :setting_aggregate_reblogs, as: :boolean, wrapper: :with_label, recommended: true
 
   - unless Setting.hide_followers_count
     .fields-group
       = f.input :setting_hide_followers_count, as: :boolean, wrapper: :with_label, glitch_only: true
 
+  .fields-group
+    = f.input :setting_norss, as: :boolean, wrapper: :with_label, label: t('admin.settings.default_norss.title'), hint: t('admin.settings.default_norss.desc_html')
+
```
Comment (Plastikmensch) [inline app/views/settings/preferences/other/show.html.haml:25]:
```suggestion
```

This setting is exposed to every user.
It belong in `app/views/admin/settings/edit.html.haml`

File: app/controllers/accounts_controller.rb [hunk 2]
```diff
@@ -24,6 +24,11 @@ def show
       format.rss do
         expires_in 1.minute, public: true
 
+        if @account&.user&.setting_norss == true
```
Comment (Plastikmensch) [inline app/controllers/accounts_controller.rb:27]:
```suggestion
        if @account&.user&.setting_norss
```
Boolean values don't have to be explicitly checked for '== true' or '== false'

File: config/locales/simple_form.en.yml [hunk 3]
```diff
@@ -205,6 +205,7 @@ en:
         setting_expand_spoilers: Always expand posts marked with content warnings
         setting_hide_network: Hide your social graph
         setting_noindex: Opt-out of search engine indexing
+        setting_norss: Opt-out of an RSS feed for your public posts
```
Comment (Plastikmensch) [inline config/locales/simple_form.en.yml:208]: I think the wording is unclear, unless you know that having an RSS feed is even a thing and what that even means.
I'm not sure what the best alternative would be though.

**Review by Plastikmensch — CHANGES_REQUESTED** [review]

File: app/views/settings/preferences/other/show.html.haml (hunk 1 above)
Comment (prplecake) [reply app/views/settings/preferences/other/show.html.haml:25]: What the issue? It's a per-user option. Checking the box would not disable rss feeds for all users on a server. 

Furthermore, the file you mentioned doesn't exist.

Comment (Plastikmensch) [reply app/views/settings/preferences/other/show.html.haml:25]: Ah, I just compared it to the hometown commit.
hometown still uses 3.5.x and a lot changed in 4.0.x.
It should probably be in `app/views/admin/settings/other/show.html.haml`  or `app/views/admin/settings/discovery/show.html.haml` even though both don't really fit.

The issue is, that that part isn't a user option, but an admin setting to set a default for \*every\* user.
Or at least that's what it's supposed to do.

Comment (prplecake) [reply app/views/settings/preferences/other/show.html.haml:25]: I see that's what the setting description did - that should change. This PR seems to only bring the per-user choice over, I missed the admin override.

Comment (Plastikmensch) [reply app/views/settings/preferences/other/show.html.haml:25]: I think letting admins change the default without having to manually edit `settings.yml` is a good thing.
Sadly upstream also removed that option for noindex.

Comment (prplecake) [reply app/views/settings/preferences/other/show.html.haml]: [10372c8](https://github.com/glitch-soc/mastodon/pull/2002/commits/10372c8d331ca5adf08a9b72790b684d5e19f613) resolves this.

**Review by Plastikmensch — APPROVED** [review]
Code looks good to me and works.
Personally I would prefer this being a glitch-only setting and I still have issues with the wording, but maybe a hint added to the user setting would help with that.

**Comment by Plastikmensch** [comment]
> The code looks ok (but the tests need updating) although I think the settings should be labelled glitch-only and the translation strings should be in `locales-glitch`.

I agree.

> But more fundamentally, as I explained in [#1936 (comment)](https://github.com/glitch-soc/mastodon/issues/1936#issuecomment-1344751813), I'm not sure a setting just for RSS makes a lot of sense: if the reason for this feature is prevent people who do not follow you to easily keep up with your public/unlisted posts… then they can as easily use the REST API. So an option that would make more sense I think would be a setting to make your list of posts unavailable while not logged in, although that can still be sidestepped by querying another server where you have followers.

Yes, but putting `.rss` to an account link is trivial, while using the REST API or sidestepping by quering another server adds inconvenience. There are certainly tools for that and this is not a be-all end-all solution, but I still think this is better than nothing.

**Comment by Plastikmensch** [comment]
> I agree that querying another server is a somewhat significant inconvenience, but querying the REST API is only marginally more inconvenient for someone to do manually, and arguably even easier to do programmatically than accessing RSS. If the main concern is qoto or qoto-like people who have built a tool, they could just as easily change it to use the REST API.

I think we're on the same page regarding that, this won't stop tools designed for block evasion, but it will stop individuals just subscribing to the RSS feed instead.
Again, "better than nothing".

**Pull request closed without merging**